sklearn/naive_bayes.py

```diff
@@ -606,12 +618,12 @@ def _check_alpha(self):
             if np.min(alpha) < 0:
                 raise ValueError("All values in alpha must be greater than 0.")
         alpha_min = 1e-10
-        if np.min(alpha) < alpha_min:
+        if np.min(alpha) < alpha_min and not _force_alpha:
             warnings.warn(
                 "alpha too small will result in numeric errors, setting alpha ="
-                f" {alpha_min:.1e}"
+                f" {alpha_min:.1e}. Use set_alpha = True to keep alpha unchanged."
             )
-            return np.maximum(alpha, alpha_min)
+            return alpha if _force_alpha else np.maximum(alpha, alpha_min)
```

Comment:
This statement is reached only when `_force_alpha` is `False` based on the `... and not _force_alpha` condition a few lines above:

```suggestion
            return np.maximum(alpha, alpha_min)
```